app/controllers/MissionController.scala

```diff
@@ -108,20 +128,44 @@ class MissionController @Inject() (implicit val env: Environment[User, SessionAu
     )
   }
 
+  def ~=(x: Double, y: Double, precision: Double) = { // Approximate equality check for Doubles
+    if ((x - y).abs < precision) true else false
+  }
+
   /**
     *
     * @param userId
     * @param regionId
     */
   def updatedUnmarkedCompletedMissionsAsCompleted(userId: UUID, regionId: Int): Unit = {
+    val completedMissions = MissionTable.selectCompletedMissionsByAUser(userId, regionId)
+    val hasCompletedOriginal1000FtMission = completedMissions.exists(m => 
+      m.coverage match {
+        case None => ~=(m.distance_ft.getOrElse(Double.PositiveInfinity), 1000.0, 0.10)
+        case _ => false
+      }
+    )
     val missions = MissionTable.selectIncompleteMissionsByAUser(userId, regionId)
     val streets = StreetEdgeTable.selectStreetsAuditedByAUser(userId, regionId)
     val CRSEpsg4326 = CRS.decode("epsg:4326")
     val CRSEpsg26918 = CRS.decode("epsg:26918")
     val transform = CRS.findMathTransform(CRSEpsg4326, CRSEpsg26918)
     val completedDistance_m = streets.map(s => JTS.transform(s.geom, transform).getLength).sum
 
-    val missionsToComplete = missions.filter(_.distance.getOrElse(Double.PositiveInfinity) < completedDistance_m)
+    // If User has audited more distance than a mission's distance in a particular region, mark the mission as completed
+    val missionsToComplete = missions.filter(m =>
+      m.distance.getOrElse(Double.PositiveInfinity) < completedDistance_m &&
+      !(m.coverage match {
+        case None => (
+          (hasCompletedOriginal1000FtMission && ~=(m.distance_ft.getOrElse(Double.PositiveInfinity), 500.0, 0.10)) ||
+          ~=(m.distance_ft.getOrElse(Double.PositiveInfinity), 1000.0, 0.10)
+        )
+        case _ => (
+          hasCompletedOriginal1000FtMission && ~=(m.distance_ft.getOrElse(Double.PositiveInfinity), 1000.0, 0.10)
+        ) // Don't add the new 500-ft and 1000-ft missions if the user has already completed the old 1000-ft mission
```

Comment:
Always add the comments above the blocks of code instead of below. This comment could also be a bit more clear.